lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -1702,6 +1702,7 @@ public function get_collection_params() {
 		$params['filter'] = array(
 			'description'       => __( 'Use WP Query arguments to modify the response; private query vars require appropriate authorization.' ),
 		);
+		$params['filter[posts_per_page]'] = $params['per_page'];
```

Comment:
Actually my first idea was `unset( $args['filter']['posts_per_page'] );` too.
But I can't figure it out that how do we handle other pagination params.

Following should be removed too?

* nopaging (boolean) - show all posts or use pagination. Default value is 'false', use paging.
* posts_per_page (int) - number of post to show per page (available since Version 2.1, replaced showposts parameter). Use 'posts_per_page'=>-1 to show all posts (the 'offset' parameter is ignored with a -1 value). Set the 'paged' parameter if pagination is off after using this parameter. Note: if the query is in a feed, wordpress overwrites this parameter with the stored 'posts_per_rss' option. To reimpose the limit, try using the 'post_limits' filter, or filter 'pre_option_posts_per_rss' and return -1
* posts_per_archive_page (int) - number of posts to show per page - on archive pages only. Over-rides posts_per_page and showposts on pages where is_archive() or is_search() would be true.
* offset (int) - number of post to displace or pass over. Warning: Setting the offset parameter overrides/ignores the paged parameter and breaks pagination (Click here for a workaround). The 'offset' parameter is ignored when 'posts_per_page'=>-1 (show all posts) is used.
* paged (int) - number of page. Show the posts that would normally show up just on page X when using the "Older Entries" link.
* page (int) - number of page for a static front page. Show the posts that would normally show up just on page X of a Static Front Page.
* ignore_sticky_posts (boolean) - ignore post stickiness (available since Version 3.1, replaced caller_get_posts parameter). false (default): move sticky posts to the start of the set. true: do not move sticky posts to the start of the set.